### PR TITLE
Fix fatal error handling in V3 API

### PR DESCRIPTION
**Pull request by simolus3**
This is related to https://github.com/isoos/postgresql-dart/issues/119.

This fixes some of the problems @osaxma has discovered and reported in that issue:

- We sometimes set the `_pending` operation without properly clearing it when we're done.
- When we receive a fatal error message, we sometimes continue to wait for a `ReadyForQueryMessage` that we'll never receive.
- The awkward setup with two completers in `_sendAndWaitForQuery` can lead to errors being swallowed.

This PR should fix all of them, so I've migrated the fixme test into the main v3 test file.

### CODE REVIEW

**Comment by simolus3** [comment]
There is some duplicate code in the error handling logic now, I'll look into ways to improve that as well.

File: lib/src/v3/connection.dart [hunk 1]
```diff
@@ -73,40 +73,61 @@ abstract class _PgSessionBase implements PgSession {
...
+          }
+        } else if (message is ReadyForQueryMessage) {
+          // This is the message we've been waiting for, the server is signalling
+          // that it's ready for another message - so we can release the lock.
+          doneWithOperation.complete();
         } else {
-          completer.completeError(
-              StateError('Unexpected message $message'), trace);
+          result =
+              Result.error(StateError('Unexpected message $message'), trace);
```
Comment (isoos) [inline lib/src/v3/connection.dart:121]: `doneWithOperation` seems to be not completed in this case. Wouldn't it hang in this case?

**Review by isoos — CHANGES_REQUESTED** [review]

File: lib/src/v3/connection.dart (hunk 1 above)
Comment (isoos) [reply lib/src/v3/connection.dart:121]: The tests are now stuck for 4h+ hours running... please check if this is related to that.

Comment (simolus3) [reply lib/src/v3/connection.dart:122]: Sorry about the timeouts, I couldn't reproduce them with podman somehow but they happen with Docker... They should be fixed now, but I've also reduced the timeout for the whole CI to catch them earlier.

If we get an unexpected message, there's either a serious bug in the v3 implementation or something very wrong the postgres connection. I think the best course might be to just close the connection in that case, we can't really expect to recover if we have no idea what's going on.

File: .github/workflows/dart.yml [hunk 2]
```diff
@@ -14,6 +14,7 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+    timeout-minutes: 10
```
Comment (isoos) [inline .github/workflows/dart.yml:17]: Do we need this? The test runs in 2.5 minutes or so...

**Review by isoos — APPROVED** [review]
I'm following maybe 50% of what happens here with the states and the code, but overall looks reasonable. Thanks for looking into it!

**Pull request merged**